python/jupyterlab-chat/src/index.ts

```diff
@@ -298,11 +299,24 @@ const docFactories: JupyterFrontEndPlugin<IChatFactory> = {
 
     // Handle state restoration.
     if (restorer) {
+      // Promise that resolve when the openChat command is ready.
+      const openCommandReady = new PromiseDelegate<void>();
+      const commandChanged = () => {
+        if (app.commands.hasCommand(CommandIDs.openChat)) {
+          openCommandReady.resolve();
+          app.commands.commandChanged.disconnect(commandChanged);
+        }
+      };
+      app.commands.commandChanged.connect(commandChanged);
+
       void restorer.restore(tracker, {
-        command: 'docmanager:open',
-        args: panel => ({ path: panel.context.path, factory: FACTORY }),
-        name: panel => panel.context.path,
-        when: app.serviceManager.ready
+        command: CommandIDs.openChat,
+        args: widget => ({
+          filepath: widget.model.name ?? '',
+          inSidePanel: widget instanceof ChatWidget
+        }),
+        name: widget => widget.model.name,
```

Comment:
I spent about 15 min reviewing how this PR works, and I'd like to confirm my understanding. This works because:

- The tracker maintains a list of all chat widgets, now including widgets in the side panel.
- This list is ordered by usage, i.e. the last item in this list is the most recently opened chat widget.
- When this plugin is activated, the layout restorer iterates through this list and opens each chat widget, automatically focusing the most recently opened chat widget.

Is this correct?

Reply:
Yes, this is how it work. The chats are restored in order, whether they are in the side panel or the main area.